File: src/main/java/net/auoeke/headsdowndisplay/HandlingKeyBinding.java
package net.auoeke.headsdowndisplay;

import net.minecraft.client.option.KeyBinding;

public class HandlingKeyBinding extends KeyBinding {
    private final Runnable handler;

    public HandlingKeyBinding(String translationKey, Runnable handler) {
        super(translationKey, -1, "key.category.headsdowndisplay");

        this.handler = handler;
    }

    @Override
    public void setPressed(boolean pressed) {
        if (pressed && !this.isPressed()) {
            this.handler.run();
        }

        super.setPressed(pressed);
    }
}

File: source/headdowndisplay/HandlingKeyBinding.kt
package headdowndisplay

import net.minecraft.client.option.KeyBinding
import java.lang.Runnable

class HandlingKeyBinding(translationKey: String, private val handler: Runnable) : KeyBinding(translationKey, -1, "key.category.head-down-display") {
    override fun setPressed(pressed: Boolean) {
        if (pressed && !this.isPressed) {
            handler.run()
        }

        super.setPressed(pressed)
    }
}
